frontend/alert/components/Timeline.tsx

```diff
@@ -0,0 +1,297 @@
+  isTime?: boolean | false;
+}
+const InfoLabel = styled.div<InfoLabelProps>`
+  font-size: 0.9375rem;
+  color: rgba(40,40,40,1);
+  height: 0.875rem;
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  justify-self: ${({isTime}) => isTime ? "end" : "start"};
+`;
+
+const convertTime = (timeString) => {
+  let d = new Date(timeString);
+  
+  //format: ["MM:DD", "HR:MIN pm/am"]
+  return [
+    d.toLocaleDateString('en-US', {month:"numeric", day: "numeric"}), 
+    d.toLocaleTimeString('en-US', { hour12: true, hour: "numeric", minute: "numeric"}).toLowerCase()
+  ]
+}
+
+const getMonthDay = (timeString) => {
+  let d = new Date(timeString);
+  return d.toLocaleDateString('en-US', {month:"numeric", day: "numeric"})
+}
+
+const formatData = (courseData) => {
+  // convert the course status data into an array of 
+  // length 3 arrays with the status data object, the current status at each created_at data point
+  // and if the date is different
+  let formattedData = courseData.reduce((ans, item, index) => { 
+
+    const sameDate = index == 0 || getMonthDay(courseData[index]["created_at"]) == getMonthDay(courseData[index - 1]["created_at"])
+
+    if (item["old_status"] == "C" && item["new_status"] == "O"){
+
+      ans.push([item, "opened", sameDate]);
+    } else {
+      ans.push([item, "closed", sameDate]);
+    }
+    
+    return ans;
+    }, [])
+
+    return formattedData
+}
+
+const timeInHours = (timeString) => {
+  let d = new Date(timeString);
+  return d.getHours();
+}
+
+const createTimelineEle = (courseData, index) => {
+  let prevTime = courseData[index - 1][0]["created_at"]
+  let currTime = courseData[index][0]["created_at"]
+
+  let segLength = Math.min(Math.round(MIN_SEGMENT_LENGTH + (Math.abs(timeInHours(prevTime) - timeInHours(currTime)) * MULTIPLIER)), MAX_SEGMENT_LENGTH)
+
+  let circle = <> 
+    <InfoLabel>{!courseData[index][2] && convertTime(currTime)[0]}</InfoLabel>
+    <Circle open={courseData[index][1] == "opened"}><FontAwesomeIcon icon={faCircle}/></Circle>
+    <InfoLabel isTime={true}>{convertTime(currTime)[1]}</InfoLabel>
+  </>
+
+  if (index - 1 == 0) {
+    return [<>
+        <InfoLabel>{convertTime(prevTime)[0]}</InfoLabel>
+        <Segment open={courseData[index - 1][1] == "opened"} length={segLength}/>
+        <InfoLabel isTime={true}>{convertTime(prevTime)[1]}</InfoLabel>
+        </>, circle];
+  } else {
+    return [<>
+        <div></div>
+        <Segment open={courseData[index - 1][1] == "opened"} length={segLength}/>
+        <div></div>
+          </>, circle];
+  }
+}
+
+interface TimelineProps {
+  courseCode: string | null;
+  setTimeline: React.Dispatch<React.SetStateAction<string | null>>;
+}
+
+const Timeline = ({
+  courseCode,
+  setTimeline}: TimelineProps) => {
+  
+  const [courseStatusData, setCourseStatusData] = useState([]);
+  const [loaded, setLoaded] = useState<boolean>(false);
+
+  useEffect( () => {
+
+      if (courseCode == null) {
```

Comment:
Instead of checking `courseCode == null` (you do this in a few other places too), it's better to check `!courseCode`, since this takes care of both null and undefined without requiring use of loose equality (undefined == null will return true, which can lead to ambiguous code sometimes).